### PR TITLE
feat: add a new 'Automate | Continuous Integration' section to the documentation

### PR DESCRIPTION
**Pull request by pepicrft**
In a recent conversation with a developer that uses Tuist, I got as a feedback that we have no guidance on how to use Tuist in continuous integration. This PR adss a new section to the documentation, `Develop > Integrate`, and a page that documents how to install and run Tuist in some popular CI providers.

### CODE REVIEW

File: docs/docs/guides/develop/integrate.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  push:
+    branches:
+      - main
+jobs:
+  lint:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: jdx/mise-action@v2
+      - run: tuist build
```
Comment (Sherlouk) [inline docs/docs/guides/develop/integrate.md:92]: Minor nit pick, and something you'd likely want to edit yourself as there are other examples in this file -- but confusing mismatch of terms here for what is demonstrating a build but talking about linting and testing.

```suggestion
name: Build Application
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
jobs:
  build:
    runs-on: macos-latest
    steps:
      - uses: actions/checkout@v3
      - uses: jdx/mise-action@v2
      - run: tuist build
```

File: docs/docs/guides/develop/integrate.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+---
+title: Integrate
+titleTemplate: ":title | Develop | Tuist"
+description: Learn how to use Tuist in your CI workflows.
+---
+
+# Integrate
```
Comment (fortmarek) [inline docs/docs/guides/develop/integrate.md:7]: I would call this `Continous integration` instead. `Integrate` is too generic and in the guide, we only reference Continuous integration.

File: docs/docs/guides/develop/integrate.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+Integration is the automated process of ensuring that changes are valid. It is often performed continuously (CI) and is a key part of the development process. Configuration is typically handled through a file in the repository, which the integration platform uses to determine how to integrate the changes.
```
Comment (fortmarek) [inline docs/docs/guides/develop/integrate.md:9]: If we use Continuous integration, I don't think we need to explain what that is anymore. Something closer to mise: https://mise.jdx.dev/continuous-integration.html

Comment (pepicrft) [reply docs/docs/guides/develop/integrate.md:9]: I adjusted it accordingly.

File: docs/docs/guides/develop/integrate.md (hunk 2 above)
Comment (pepicrft) [reply docs/docs/guides/develop/integrate.md:7]: I renamed it to `Continuous Integration`, and placed it under a section called `Automate`, which leaves space for the future work that's coming:

File: docs/docs/guides/develop/automate/continuous-integration.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+description: Learn how to use Tuist in your CI workflows.
+---
+
+# Continuous Integration (CI)
+
+You can use Tuist in [continuous integration](https://en.wikipedia.org/wiki/Continuous_integration) environments. The following sections provide examples of how to do this on different CI platforms.
+
+## Examples
+
+To run Tuist commands in your CI workflows, you’ll need to install it in your CI environment. The following sections provide examples of how to do this on different CI platforms.
```
Comment (fortmarek) [inline docs/docs/guides/develop/automate/continuous-integration.md:13]:
```suggestion
To run Tuist commands in your CI workflows, you’ll need to install it in your CI environment.
```

The "following sections..." appears twice here (it's in the `CI` header as well)

**Review by fortmarek — APPROVED** [review]

**Pull request merged**